### PR TITLE
Added Azure.MariaDB.FirewallIPRange

### LINKED ISSUE #1856: Azure Database for MariaDB servers should restrict an excessive number of permitted IP addresses
# Rule request

## Suggested rule change

Azure Database for MariaDB servers should restrict an excessive number of firewall IP permitted IP addresses.

Security pillar for this one.

## Applies to the following

The rule applies to the following:

- Resource type: **[Microsoft.DBforMariaDB/servers]**

## Additional context

[Create and manage Azure Database for MariaDB firewall rules](https://learn.microsoft.com/azure/mariadb/howto-manage-firewall-portal)
[Template reference](https://learn.microsoft.com/azure/templates/microsoft.dbformariadb/servers)

### PR DESCRIPTION
**Pull request by BenjaminEngeset**
## PR Summary

Fixes #1856

Added Azure.MariaDB.FirewallIPRange rule.

## PR Checklist

- [x] PR has a meaningful title
- [ ] Summarized changes
- [ ] Change is not breaking
- [ ] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [ ] Unit tests created/ updated
  - [ ] Rule documentation created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by BenjaminEngeset** [comment]
Hey @BernieWhite 

I need some help. I've tried to utilize some functions and rule design created for previous rules. I am not getting the expected behaviour as intended and I need help to understand why.

- I am getting failed results for `server-B` and `server-C` where both of the objects property `Public` has a value of 1 and I'm using `LessOrEqual` method with an expected integer value of less than or equal to `10<=`.

`The number of public IP addresses permitted (1) exceeded 10.`
`The number of public IP addresses permitted (1) exceeded 10.`

File: src/PSRule.Rules.Azure/rules/Azure.MariaDB.Rule.ps1 [hunk 1]
```diff
@@ -126,6 +126,14 @@ Rule 'Azure.MariaDB.FirewallRuleCount'-Ref 'AZR-000343' -Type 'Microsoft.DBforMa
     Reason($LocalizedData.DBServerFirewallRuleCount, $firewallRules.Length, 10).PathPrefix('resources')
 }
 
+# Synopsis: Determine if there is an excessive number of permitted IP addresses.
+Rule 'Azure.MariaDB.FirewallIPRange' -Ref 'AZR-000344' -Type 'Microsoft.DBforMariaDB/servers' -Tag @{ release = 'GA'; ruleSet = '2022_12'; } {
+    $summary = GetIPAddressSummary
+
+    $Assert.LessOrEqual($summary, 'Public', 10).
```
Comment (BernieWhite) [inline src/PSRule.Rules.Azure/rules/Azure.MariaDB.Rule.ps1:133]: @BenjaminEngeset There seems to be a bug with the `LessOrEqual` method not handling uint64 types correctly. Try this.

```suggestion
    [int]$public = [int]$summary.Public
    $Assert.LessOrEqual($public, '.', 10).
```

**Review by BernieWhite — COMMENTED** [review]
@BenjaminEngeset See the comments below.

**Comment by BenjaminEngeset** [comment]
@BernieWhite 

You were correct. Casting it directly to int type did the trick.

Ready for review btw.

**Review by BernieWhite — APPROVED** [review]
Thanks @BenjaminEngeset. Great work. All good to merge.

**Pull request merged**